src/hooks/useWebR.tsx

```diff
@@ -251,7 +253,51 @@ webr::shim_install()
 }
 
 .webr_pkg_cache <- list()
+
+.mount_vfs_images <- function() {
+  metadata_url <- glue::glue("{.base_url}packages/metadata.rds")
+  metadata_path <- glue::glue("/shinylive/webr/packages/metadata.rds")
+
+  # Attempt this download quietly, if no metadata exists we can still continue
+  found <- webr::eval_js(glue::glue("
+    var xhr = new XMLHttpRequest();
+    xhr.open('HEAD', '{metadata_url}', false);
+    xhr.send();
+    (xhr.status >= 200 && xhr.status < 300)
+  "))
+  if (found) {
+    download.file(metadata_url, metadata_path, quiet = TRUE)
+  }
+
+  if (file.exists(metadata_path)) {
+    metadata <- readRDS(metadata_path)
+    lapply(metadata, function(data) {
+      name <- data$name
+      path <- data$path
+      cached <- data$cached
+      mountpoint <- glue::glue("/shinylive/webr/packages/{name}")
+
+      # Mount the virtual filesystem image, unless we already have done so
+      if (cached && !file.exists(mountpoint)) {
```

Comment:
This is a reuse of shinylive-r's `cached` metadata property. It originally means "shinylive-r has successfully downloaded the correct version of this package and included it in the app bundle." We use it here as a signal that the associated `.data` file exists and we can mount it.

I guess it's not a great variable name, since this confusion is possible. I think I will change it to be:
```r
available <- data$cached
```

and 

```r
if (available && !file.exists(mountpoint)) {
```

to make it clearer.